dietpi/dietpi-software

```diff
@@ -30,6 +30,29 @@ Available commands:
 	G_CHECK_ROOTFS_RW
 	G_INIT
 	# Import DietPi-Globals ---------------------------------------------------------------
+  
+  # Configuration Creation
+CONFIG_CREATE(){
+  [[ $G_PROGRAM_NAME ]] || local G_PROGRAM_NAME='G_CONFIG_CREATE'
+  local file=$1
+  local service=$2
+  local timeout=15 
+  (( $G_HW_ARCH > 2 )) || timeout=20 
+  G_EXEC_DESC="Starting {$service^} to pre-create config file" G_EXEC systemctl start "$service" 
+  local i=0 
+  until [[ -f "$file" || $i -ge $timeout ]] 
+  do 
+   ((i++)) 
+   G_DIETPI-NOTIFY -2 "Waiting for {$service^} config file to be created ($i/$timeout)" 
+   sleep 1 
+  done 
+  sleep 1 
+  G_EXEC_NOHALT=1 G_EXEC systemctl stop "$service" 
+  sleep 1
+  if [[ ! -f "$file" ]]
+  then
+    G_DIETPI-NOTIFY 1 "Waiting for {$service^} config file to be created timed out, skipping pre-configuration"
+}
```

Comment:
A closing `fi` is missing here.

Also: We need an explicit return code when the file has not been created in the end (timed out) so that configuration steps can be done or skipped conditionally.